### DISCUSSION TITLE
Asserting on metrics in Tests

### DISCUSSION THREAD

**Discussion by cpaika** [open-telemetry/opentelemetry-rust#2062] [General]
Sometimes while writing unit/integration tests you may want to assert on certain metrics being set - for example, you send a few HTTP queries and assert the metrics are being incremented correctly.  

If I have a metrics object, I can access counters but I can't seem to get the actual underlying value of the counter as a i64/u64/etc to be able to assert it has a specific value.  Is there a way to do this I'm missing?  If not can it be tracked as a feature request?

**Comment by utpilla** [comment]
For Counters, you could downcast `data` field of the `Metric` [struct](https://github.com/open-telemetry/opentelemetry-rust/blob/7d47bf213dac15c18c9ae9dcfba5d502e4fd4282/opentelemetry-sdk/src/metrics/data/mod.rs#L34-L44) to a `Sum` [struct](https://github.com/open-telemetry/opentelemetry-rust/blob/7d47bf213dac15c18c9ae9dcfba5d502e4fd4282/opentelemetry-sdk/src/metrics/data/mod.rs#L75-L84) which would give you access to the actual values from `Vec<DataPoints>`.

You could reference the [code](https://github.com/open-telemetry/opentelemetry-rust/blob/7d47bf213dac15c18c9ae9dcfba5d502e4fd4282/opentelemetry-stdout/src/metrics/exporter.rs#L114-L135) in stdout exporter. It does the same kind of data transformations to print out the metric values.

**Reply by cpaika** [reply]
Awesome, thank you!
